pkg/firewall/firewall_test.go

```diff
@@ -1,162 +1,332 @@
 package firewall
 
 import (
-	"github.com/keep-network/keep-core/pkg/chain/local_v1"
-	"github.com/keep-network/keep-core/pkg/operator"
-	"math/big"
+	"fmt"
 	"testing"
 	"time"
 
 	"github.com/keep-network/keep-common/pkg/cache"
+	"github.com/keep-network/keep-core/pkg/chain/local_v1"
+	"github.com/keep-network/keep-core/pkg/internal/testutils"
+	"github.com/keep-network/keep-core/pkg/operator"
 )
 
-var minimumStake = big.NewInt(1000)
-var cachingPeriod = time.Second
+const cachingPeriod = time.Second
 
-func TestHasMinimumStake(t *testing.T) {
-	stakeMonitor := local_v1.NewStakeMonitor(minimumStake)
-	policy := &minimumStakePolicy{
-		stakeMonitor:        stakeMonitor,
+func TestValidate_OperatorNotRecognized_NoApplications(t *testing.T) {
+	policy := &anyApplicationPolicy{
+		applications:        []Application{},
 		positiveResultCache: cache.NewTimeCache(cachingPeriod),
 		negativeResultCache: cache.NewTimeCache(cachingPeriod),
 	}
 
-	_, remotePeerOperatorPublicKey, err := operator.GenerateKeyPair(local_v1.DefaultCurve)
+	_, peerOperatorPublicKey, err := operator.GenerateKeyPair(
+		local_v1.DefaultCurve,
+	)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	err = stakeMonitor.StakeTokens(remotePeerOperatorPublicKey)
+	err = policy.Validate(peerOperatorPublicKey)
+	testutils.AssertErrorsEqual(t, errNotRecognized, err)
+}
+
+func TestValidate_OperatorNotRecognized_MultipleApplications(t *testing.T) {
+	_, peerOperatorPublicKey, err := operator.GenerateKeyPair(
+		local_v1.DefaultCurve,
+	)
+
+	// Use two application: the first returns error during check, the second
+	// does not recognize the operator
+	application := newMockApplication()
+	application.setIsRecognized(peerOperatorPublicKey, result{
+		isRecognized: false,
+		err:          fmt.Errorf("dummy error"),
+	})
+	policy := &anyApplicationPolicy{
+		applications: []Application{
+			newMockApplication(),
+			newMockApplication()},
+		positiveResultCache: cache.NewTimeCache(cachingPeriod),
+		negativeResultCache: cache.NewTimeCache(cachingPeriod),
+	}
+
 	if err != nil {
```

Comment:
This one should be placed right after `GenerateKeyPair` call at line 35.

Reply:
Ok, now I get what you mean :)
Done in 75be1d4a4618a1dd55a965be01d67e24475ed93b